src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -29,6 +29,121 @@ class OrchestratorCli(orchestrator.OrchestratorClientMixin, MgrModule):
         {'name': 'orchestrator'}
     ]
 
+    def __init__(self, *args, **kwargs):
+        super(OrchestratorCli, self).__init__(*args, **kwargs)
+        self.ident = set([])
+        self.fault = set([])
+        self.devs = {}
+        self._load()
+        self._refresh_health()
+
+    def _load(self):
+        active = self.get_store('active_devices')
+        if active:
+            decoded = json.loads(active)
+            self.ident = set(decoded.get('ident', []))
+            self.fault = set(decoded.get('fault', []))
+        self.log.debug('ident %s, fault %s' % (self.ident, self.fault))
+
+    def _save(self):
+        encoded = json.dumps({
+            'ident': list(self.ident),
+            'fault': list(self.fault),
+            })
+        self.set_store('active_devices', encoded)
+
+    def _refresh_health(self):
+        h = {}
+        if self.ident:
+            h['DEVICE_IDENT_ON'] = {
+                'severity': 'warning',
+                'summary': '%d devices have ident light turned on' % len(
+                    self.ident),
+                'detail': ['%s ident light enabled' % d for d in self.ident]
+            }
+        if self.fault:
+            h['DEVICE_FAULT_ON'] = {
+                'severity': 'warning',
+                'summary': '%d devices have fault light turned on' % len(
+                    self.fault),
+                'detail': ['%s fault light enabled' % d for d in self.fault]
+            }
+        self.set_health_checks(h)
+
+    def _get_devices(self):
+        d = self.get('devices')
+        dm = {}
+        for i in d['devices']:
+            dm[i['devid']] = i['location']
+        return dm
```

Comment:
This will be empty if you happen to be testing on VMs with virtio disks (device_id is never set for those, so self.get('devices') returns an empty list). If I switch my test VMs to using SATA or SCSI, then device_id is populated, but they're not unique across the cluster, e.g.:

```
# ceph device ls
DEVICE                HOST:DEV                                DAEMONS                     LIFE EXPECTANCY 
QEMU_HARDDISK_QM00005 data1:sda data2:sda data3:sda           osd.0 osd.2 osd.5                           
QEMU_HARDDISK_QM00007 data1:sdb data2:sdb data3:sdb data4:sda osd.1 osd.16 osd.3 osd.4                    
QEMU_HARDDISK_QM00009 data1:sdc data2:sdc data3:sdc data4:sdb osd.22 osd.6 osd.7 osd.8                    
QEMU_HARDDISK_QM00011 data1:sdd data2:sdd data3:sdd data4:sdc osd.10 osd.11 osd.12 osd.9                  
QEMU_HARDDISK_QM00013 data1:sde data2:sde data3:sde data4:sdd osd.13 osd.14 osd.15 osd.20                 
QEMU_HARDDISK_QM00015 data1:sdf data2:sdf data3:sdf data4:sde osd.17 osd.18 osd.19 osd.23                 
QEMU_HARDDISK_QM00017 data4:sdf                               osd.21                                      
```

Now, I know there's no point trying to test blinky lights on VMs ;-) but this does make me wonder, could we have real clusters where device_ids are not globally unique, or cannot be determined, and thus will be invisible in this context?

Reply:
Yep.  The get_device_id() in common/blkdev.cc is where we try to map a devname ("sda") to a globally unique vendor/model/serial style string.  VMs hide enough that I think it is probably a lost cause, but maybe we can special case something like the qemu device names to include a vm id or something so that they are at least unique.